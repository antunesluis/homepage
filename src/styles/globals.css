@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  /* Light theme */
  --color-background: #fafafa;
  --color-foreground: #18181b;

  /* Primary colors - Blue/Indigo palette */
  --color-primary: #4f46e5;
  --color-primary-hover: #4338ca;
  --color-primary-muted: #e0e7ff;

  /* Secondary colors - Teal palette */
  --color-secondary: #0d9488;
  --color-secondary-hover: #0f766e;
  --color-secondary-muted: #ccfbf1;

  /* Accent colors */
  --color-accent: #f43f5e;
  --color-accent-hover: #e11d48;
  --color-accent-muted: #ffe4e6;

  /* Gray scale */
  --color-muted: #f4f4f5;
  --color-border: #e4e4e7;
  --color-border-hover: #d4d4d8;

  /* Surface colors for cards and elevated elements */
  --color-surface: #ffffff;
  --color-surface-hover: #f8fafc;
}

.dark {
  /* Dark theme */
  --color-background: #09090b;
  --color-foreground: #fafafa;

  /* Primary colors - adjusted for dark mode */
  --color-primary: #6366f1;
  --color-primary-hover: #818cf8;
  --color-primary-muted: #1e1b4b;

  /* Secondary colors - adjusted for dark mode */
  --color-secondary: #14b8a6;
  --color-secondary-hover: #2dd4bf;
  --color-secondary-muted: #134e4a;

  /* Accent colors - adjusted for dark mode */
  --color-accent: #fb7185;
  --color-accent-hover: #fda4af;
  --color-accent-muted: #4c1d24;

  /* Gray scale - darker variants */
  --color-muted: #27272a;
  --color-border: #3f3f46;
  --color-border-hover: #52525b;

  /* Surface colors for dark mode */
  --color-surface: #18181b;
  --color-surface-hover: #27272a;
}

body {
  @apply bg-background text-foreground;
  transition:
    background-color 0.3s ease-in-out,
    color 0.3s ease-in-out;
}
