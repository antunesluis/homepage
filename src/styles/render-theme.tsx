import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'next-themes';

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider attribute="class" defaultTheme="light" enableSystem={false}>
      {children}
    </ThemeProvider>,
  );
};
