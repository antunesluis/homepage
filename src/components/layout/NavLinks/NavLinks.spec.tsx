import { screen } from '@testing-library/react';
import { NavLinks } from '.';
import { renderTheme } from '@/styles/render-theme';

describe('<NavLinks />', () => {
  it('should render all navigation items', () => {
    renderTheme(<NavLinks />);

    const navItems = ['Home', 'Projetos', 'Sobre', 'Contato'];
    navItems.forEach((item) => {
      expect(screen.getByText(item)).toBeInTheDocument();
    });
  });

  it('should have correct href attributes', () => {
    renderTheme(<NavLinks />);

    const navLinks = {
      Home: '/',
      Projetos: '/projects',
      Sobre: '/about',
      Contato: '/contact',
    };

    Object.entries(navLinks).forEach(([text, href]) => {
      expect(screen.getByText(text).closest('a')).toHaveAttribute('href', href);
    });
  });

  it('should match the snapshot', () => {
    const { container } = renderTheme(<NavLinks />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
