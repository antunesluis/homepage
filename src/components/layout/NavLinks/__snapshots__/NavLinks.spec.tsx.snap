// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NavLinks /> > should match the snapshot 1`] = `
<nav
  class="hidden md:flex items-center gap-1"
>
  <a
    class="px-4 py-2 relative rounded-lg text-sm transition-colors duration-200 hover:text-foreground/80 text-foreground font-medium"
    href="/"
  >
    Home
  </a>
  <a
    class="px-4 py-2 relative rounded-lg text-sm transition-colors duration-200 hover:text-foreground/80 text-foreground/60"
    href="/projects"
  >
    Projetos
  </a>
  <a
    class="px-4 py-2 relative rounded-lg text-sm transition-colors duration-200 hover:text-foreground/80 text-foreground/60"
    href="/about"
  >
    Sobre
  </a>
  <a
    class="px-4 py-2 relative rounded-lg text-sm transition-colors duration-200 hover:text-foreground/80 text-foreground/60"
    href="/contact"
  >
    Contato
  </a>
</nav>
`;
